src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -138,6 +146,32 @@ def validate(self, pdict=None):
                 return False, msg
         return True, msg
 
+    def validateTransitionRecord(self, records):
+        """
+        Validate given transition records
+
+        :param records: transition records
+        :return: boolean
+        """
+        for record in records:
+            unixTime = time.time() - 1*24*60*60  # validate against yesterday timestamp
```

Comment:
we can't use 0 as it is not valid unix timestamp, and neither can use time since epoch which will not have any meaning. We can increase offset to 1 week or 1 month if you prefer but I'll not lowered it further in time. Are you agree?

Reply:
I am afraid not. We cannot constrain when a pileup object was last updated/customized. I would be happy to say that a pileup object should NOT have a timestamp older then this very moment (time.time() as I write this message). Other than that, it's risky!

In other words, I could have a pileup container that I customize to have a fraction of 0.25 tomorrow, and keep using this pileup until the year of 2026. With this code (or the 1 month), it would fail this validation, while it has a sound updateTime value.

Reply:
I'm not sure I unerstand your comments then:
- *then validating the timestamp to be within the last 24h is not correct. IMO that validation function should be [0, time now].*
- - and then you say *pileup object should NOT have a timestamp older then this very moment (time.time()*
- They contradict each other, no?

I have the following logic
```
unixTime = time.time() - 1*24*60*60  # validate against yesterday timestamp
...
if key == 'updateTime' and val < unixTime:
```
which says, that time stamp should be large than yesterday. I can change it to closer to `time.time` but I can't use `time.time` strict value as it is when we create the update, so it should have some offset to the past. In other words the update timestamp should be at present minus some offset but it does not impose any constrain on a future and neither should  be 0 till unix since epoch.

Please clarify further in form of the code snippet your logic.

Reply:
ok, I like usage of insertTime for validation, and I made corresponding changes.